### DISCUSSION TITLE
Does Anytype uses IPFS or any decentralized storage systems?

### DISCUSSION THREAD

**Discussion by warengonzaga** [anyproto/contributors#15] [Q&A] [answered]
I'm curious if Anytype uses any decentralized storage system such as IPFS.

**Comment by SamBouwer** [comment]
Yes, Anytype is using IPFS. See also the Storage paragraph on https://doc.anytype.io/d/data-and-security/data-storage-and-deletion#storage

**Comment by fuksman** [comment, marked as answer]
@warengonzaga 

And here you can find more technical details about the infrastructure layer: https://tech.anytype.io/any-sync/overview

TL;DR: IPLD (one of the IPFS components) is used for file storage only.

**Reply by warengonzaga** [reply]
Perfect, will read more!

**Reply by 2color** [reply]
@fuksman The docs also mention that you use a private IPFS network [source](https://doc.anytype.io/anytype-docs/data-and-security/data-storage-and-deletion#storage) is that actually the case? Because the [any-sync docs](https://tech.anytype.io/any-sync/overview?id=files) goes into why you developed your own sync approach rather than bitswap
 
> any-sync has its own approach for file retrieval, as opposed to [bitswap](https://docs.ipfs.tech/concepts/bitswap/), which is designed for networks with complex topologies.

**Reply by mcrakhman** [reply]
Basically we are using our own implementation of `BlockService` which asks peers directly for the blocks, where peers could be either local peers or the nodes which are registered in current network (here Anytype can have different networks).

**Reply by 2color** [reply]
@mcrakhman Thanks! 
Have you built your own request response protocol for asking peers for blocks? How do you manage these interactions?

**Reply by mcrakhman** [reply]
Yes, we have our own GRPC protocol for sending and retrieving blocks. We are trying to get blocks either from our file nodes or from local peers. 

And because we know exactly who these peers are we don't need any complicated discovery mechanism (at least for now, later we can introduce more advanced p2p discovery). File nodes are coming from node configuration and local peers are coming from MDNS.
